lldb/source/Symbol/TypeSystem.cpp

```diff
@@ -186,6 +186,11 @@ std::optional<llvm::json::Value> TypeSystem::ReportStatistics() {
   return std::nullopt;
 }
 
+CompilerDeclContext
```

Comment:
Could add a brief oxygen docstring saying: "Returns the direct parent context of specified type". Also, what happens if no such context exists (if that can happen)?

Reply:
> Could add a brief oxygen docstring saying: "Returns the direct parent context of specified type".

I'd be happy to do that as long as Jim is not opposed to the direction of `FindDirectNestedType`.

> Also, what happens if no such context exists (if that can happen)?

Half an hour ago I discovered that this can happen, and updated this patch.
See https://github.com/llvm/llvm-project/blob/434b8a30804a433526e80eb97f15dafa1815f936/lldb/source/Plugins/TypeSystem/Clang/TypeSystemClang.cpp#L2672-L2696

Reply:
Both items are done!